pkgs/test/named-hierarchy/default.nix

```diff
@@ -0,0 +1,184 @@
+{
+  lib,
+  runCommand,
+  extend,
+  callPackage,
+}:
+let
+  baseDirectory = ../../by-name;
+
+  /*
+  Directory structure:
+
+  pkgs/by-name
+  - bar
+    - bar (regular)
+  - fo
+    - @
```

Comment:
What is this `@`?

Reply:
This code in this file is still a bit drafty, but this is an example of an invalid package directory/attribute name. I will clean this code up